core/metric.go

```diff
@@ -27,26 +27,132 @@ import (
 	"github.com/intelsdi-x/snap/core/cdata"
 )
 
-type Label struct {
-	Index int    `json:"index"`
-	Name  string `json:"name"`
-}
+var (
+	// Standard Tags are in added to the metric by the framework on plugin load.
+	// STD_TAG_PLUGIN_RUNNING_ON describes where the plugin is running (hostname).
+	STD_TAG_PLUGIN_RUNNING_ON = "plugin_running_on"
+)
 
 // Metric represents a snap metric collected or to be collected
 type Metric interface {
 	RequestedMetric
 	Config() *cdata.ConfigDataNode
 	LastAdvertisedTime() time.Time
 	Data() interface{}
-	Source() string
-	Labels() []Label
 	Tags() map[string]string
 	Timestamp() time.Time
+	Description() string
+	Unit() string
+}
+
+type Namespace []NamespaceElement
+
+// String returns the string representation of the namespace with "/" joining
+// the elements of the namespace.  A leading "/" is added.
+func (n Namespace) String() string {
+	ns := ""
+	for i, x := range n {
+		if i == 0 {
+			ns = "/"
+		}
+		ns += x.Value
+		if i != len(n)-1 {
+			ns += "/"
+		}
+	}
+	return ns
+}
+
+// Strings returns an array of strings that represent the elements of the
+// namespace.
+func (n Namespace) Strings() []string {
+	return strings.Split(strings.TrimPrefix(n.String(), "/"), "/")
+}
+
+// Key returns a string representation of the namespace with "." joining
+// the elements of the namespace.
+func (n Namespace) Key() string {
+	ns := ""
+	for i, x := range n {
+		ns += x.Value
+		if i < len(n)-1 {
+			ns += "."
+		}
+	}
+	return ns
+}
```

Comment:
This is the behavior of [strings.Join()](https://golang.org/pkg/strings/#Join). Would it make sense to use that here?